### PR TITLE
Changes related to covariance R

### PR DESCRIPTION
**Pull request by msn-chowdhury**
1. The parameter Rmin has been removed from the set of mixture parameters. 

2. Also, the computation of R has been updated everywhere following the equation: R=(1-alpha) x R+alpha x (sum(diag(R))/M) x eye, where 0<alpha<=1. 

3. It came to my attention that the condition number impacts not only the initialization of the covariance matrix but also the M step of EM iterations. The reason why we missed it in the first place is because the condition number was affecting the Rmin, and the Rmin value was used in the M step. So, the impact was indirect. Thus, the description of the condition number has been slightly changed.

### CODE REVIEW

**Comment by cabouman** [comment]
Oh, this is huge! OK, the function of the condition number in the M-step is the major effect it has on the code. I will need to look at this carefully. That explains why the condition number was so important.

**Comment by cabouman** [comment]
Samin, your implementation is incorrect. What you have now is:
`# Ensure that the condition number of R is >= condition_number`
`    alpha = 1.0 / condition_number`
`    R = (1.0 - alpha) * R + alpha * np.eye(mixture.M)`

I don't know the correct python functions, but you need to do something like:
`# Ensure that the condition number of R is >= condition_number`
`    alpha = 1.0 / condition_number`
`    D = (sum(diag(R))/mixture.M) * np.eye(mixture.M)`
`    R = (1.0 - alpha) * R + alpha * D`

So in other words, the diagonal entrees of `D` should be the average of the diagonal entrees of `R`.

**Comment by msn-chowdhury** [comment]
Thank you for taking the time to look at the code. 

I have updated the computation of R using the new equation you just suggested:
R=(1-alpha) x R+alpha x (sum(diag(R))/M) x eye, where alpha=1/condition number.

After making this change, I tested the demos and the results are now correct for all different condition numbers (I tried condition_number =  1/10/100/1000). I hope this would positively impact our research results and make them less dependent on condition number.

**Comment by cabouman** [comment]
This is good, but we need to do a lot more work on this before I accept the PR.

1. You need to create a function, `ridge_regression(R, est_kind, alpha)` that implements the regularization. There are two cases to consider, when `west_kind='diag'` and when it is not.
2. We need to clean up demo_1 and the functions in sim.py. The function `GMM` should be renamed `gen_gm_samples(gm_parmas)` and its argument should be the parameters of a general GM model. The specific GM model parameters used for the demo along with any plotting should be included in the demo_1 script.
